### PR TITLE
refactor(eslint-plugin-foreman): stop using vendor-dev

### PR DESCRIPTION
**Pull request by sharvit**
Instead of using `vendor-dev`, bring the needed functionality to `eslint-plugin-foreman`.

This PR should allow us to deprecate `vendor-dev`

### CODE REVIEW

**Comment by sharvit** [comment]
This should fix katello error: https://github.com/Katello/katello/pull/8929
Foreman error is not related

**Comment by johnpmitsch** [comment]
@sharvit Looks like this needs some merge conflicts fixed, and pushing up will trigger the new tarball building actions, which would help in reviewing :smiley:

**Comment by johnpmitsch** [comment]
@sharvit what is the best way to review this? Test out linting/testing works as expected for Foreman using these packages?

File: packages/vendor-dev/package.json [hunk 1]
```diff
@@ -18,16 +18,13 @@
...
   ],
   "scripts": {
     "lint": "eslint ."
   },
   "dependencies": {
-    "@theforeman/find-foreman": "^4.15.0",
     "@theforeman/vendor-core": "^4.15.0",
-    "babel-plugin-module-resolver": "^3.2.0",
-    "eslint-import-resolver-alias": "^1.1.2"
+    "babel-plugin-module-resolver": "^3.2.0"
```
Comment (johnpmitsch) [inline packages/vendor-dev/package.json:28]: why keep "babel-plugin-module-resolver" in vendor-dev, this is still needed?

Comment (sharvit) [reply packages/vendor-dev/package.json]: I was removing the packages I stopped using in the `vendor-dev` but actually, we can safely remove the vendor-dev entirely as it is not used anymore.

**Comment by sharvit** [comment]
Thanks @johnpmitsch,

I removed the vendor-dev package and deployed it as a `next-drop-vendor-dev` version so it would be easier to test it.
(there are 3 comments because I got 3 commits in my branch 😩   )

**Comment by jeremylenz** [comment]
@sharvit what's the status of this one?  Does it need to be reassigned?

**Comment by adamruzicka** [comment]
Looks like master has drifted away a bit since this was opened. @sharvit could you please rebase?

**Comment by ezr-ondrej** [comment]
Looks :red_circle:  :thinking:

**Comment by sharvit** [comment]
I'm not sure why it fails to pull the novnc package from github but I created another PR to upgrade it and pull from npm:
https://github.com/theforeman/foreman-js/pull/373

It should probably fix the issue.

**Comment by adamruzicka** [comment]
Will someone take this over or should we close?

**Comment by MariaAga** [comment]
@Ron-Lavi @jeremylenz, can we close this?

**Pull request closed without merging**